integration-libs/cds/src/components/recent-searches/recent-searches.service.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { interval, Observable, of, ReplaySubject } from 'rxjs';
+import { concatMap, endWith, take, takeWhile } from 'rxjs/operators';
+import { WindowRef } from '@spartacus/core';
+
+@Injectable({ providedIn: 'root' })
+export class RecentSearchesService {
+  private readonly recentSearchesSource = new ReplaySubject<string[]>();
+  private apiAvailability = false;
+
+  constructor(protected winRef: WindowRef) {}
+
+  public get recentSearches$(): Observable<string[]> {
+    this.addRecentSearchesListener();
+    return this.recentSearchesSource.asObservable();
+  }
+  checkAvailability() {
+    return interval(150).pipe(
+      concatMap((_) => of((<any>this.winRef.nativeWindow).Y_TRACKING)),
+      take(5),
+      takeWhile((result: any) => !result.recentSearches),
+      endWith(true)
+    );
+  }
+
+  private addRecentSearchesListener() {
+    if (!this.apiAvailability) {
+      this.checkAvailability().subscribe((result) => {
+        if (result) {
+          const recentPhrases = (<any>(
+            this.winRef.nativeWindow
+          )).Y_TRACKING?.recentSearches?.getPhrases();
+          if (recentPhrases) {
+            this.recentSearchesSource.next(recentPhrases);
+
+            (<any>window).Y_TRACKING.recentSearches?.addListener(
```

Comment:
Probably leftover, pointing to `window` instead of `winRef`

Reply:
yeah, eagle eye :) thanks